tests/guestlog/guestlog.go

```diff
@@ -0,0 +1,267 @@
+			BeforeEach(func() {
+				originalConfig = *util.GetCurrentKv(virtClient).Spec.Configuration.DeepCopy()
+			})
+
+			AfterEach(func() {
+				tests.UpdateKubeVirtConfigValueAndWait(originalConfig)
+			})
+
+			It("should successfully start with LogSerialConsole=true", Serial, func() {
+				By("Starting a VMI")
+				vmi := tests.RunVMIAndExpectLaunch(alpineVmi, cirrosStartupTimeout)
+
+				By("Finding virt-launcher pod")
+				virtlauncherPod, err := libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(virtlauncherPod.Spec.Containers).To(HaveLen(3))
+				foundContainer := false
+				for _, container := range virtlauncherPod.Spec.Containers {
+					if container.Name == "guest-console-log" {
+						foundContainer = true
+					}
+				}
+				Expect(foundContainer).To(BeTrue())
+
+				for _, containerStatus := range virtlauncherPod.Status.ContainerStatuses {
+					if containerStatus.Name == "guest-console-log" {
+						Expect(containerStatus.State.Running).ToNot(BeNil())
+					}
+				}
+
+			})
+
+			It("should successfully start with LogSerialConsole=false", Serial, func() {
+				By("Starting a VMI")
+				alpineVmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.P(true)
+				alpineVmi.Spec.Domain.Devices.LogSerialConsole = pointer.P(false)
+				vmi := tests.RunVMIAndExpectLaunch(alpineVmi, cirrosStartupTimeout)
+
+				By("Finding virt-launcher pod")
+				virtlauncherPod, err := libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(virtlauncherPod.Spec.Containers).To(HaveLen(2))
+				foundContainer := false
+				for _, container := range virtlauncherPod.Spec.Containers {
+					if container.Name == "guest-console-log" {
+						foundContainer = true
+					}
+				}
+				Expect(foundContainer).To(BeFalse())
+			})
+
+		})
+
+		Context("fetch logs", func() {
+			var k8sClient string
+			var cirrosLogo = strings.Replace(` ____               ____  ____
+ / __/ __ ____ ____ / __ \/ __/
+/ /__ / // __// __// /_/ /\ \ 
+\___//_//_/  /_/   \____/___/ 
+   http://cirros-cloud.net
+`, "\n", "\r\n", 5)
+
+			BeforeEach(func() {
+				k8sClient = clientcmd.GetK8sCmdClient()
+				clientcmd.SkipIfNoCmd(k8sClient)
+			})
+
+			It("it should fetch logs for a running VM with oc/kubectl", func() {
+				vmi := tests.RunVMIAndExpectLaunch(cirrosVmi, cirrosStartupTimeout)
+
+				By("Finding virt-launcher pod")
+				virtlauncherPod, err := libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Executing `kubectl/oc logs` and ensure the logs are correctly ordered with no unexpected line breaks")
+
+				Eventually(func() bool {
+					outputString, _, err := clientcmd.RunCommandWithNS(virtlauncherPod.Namespace, k8sClient, "logs", virtlauncherPod.Name, "-c", "guest-console-log")
+					Expect(err).ToNot(HaveOccurred())
```

Comment:
This will likely fail the whole test case instead of being re-run on failure, see [here](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode).

Reply:
nice catch